### PR TITLE
Support for X_FORWARDED_FOR

### PR DESCRIPTION
**Pull request by marcoala**
Hello, I've added the support for the `X_FORWARDED_FOR` HTTP header (https://en.wikipedia.org/wiki/X-Forwarded-For). Unsure to mark this as bug or as new feature.

Right now if you're running on Heroku or on AWS you're behind a load balancer, this mean that the IP that you see in the `REMOTE_ADDR` is the IP address of the balancer (that in some cloud change every few minutes).

It was impossible to use the option `RESTRICTEDSESSIONS_REMOTE_ADDR_KEY` since `X_FORWARDED_FOR` maybe just a a ip address or a list of ip addresses in case of multiple proxy in the architectures.

### CODE REVIEW

**Comment by marcoala** [comment]
@erikr do you've any feedback on this one? Has been open for a while, happy to discuss all your concern.

File: requirements-test.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-django>=1.8.13
+django==1.8.13
```
Comment (erikr) [inline requirements-test.txt]: We want to test against the latest Django, not a particular version.

File: restrictedsessions/middleware.py [hunk 2]
```diff
@@ -22,15 +22,25 @@ def process_request(self, request):
             return
 
         # Short circuit for option to require authed users
+        user = getattr(request, 'user', None)
```
Comment (erikr) [inline restrictedsessions/middleware.py]: Why was this line moved?

File: restrictedsessions/middleware.py [hunk 3]
```diff
@@ -22,15 +22,25 @@ def process_request(self, request):
...
                 return
 
         # Extract remote IP address for validation purposes
-        remote_addr_key = getattr(settings, 'RESTRICTEDSESSIONS_REMOTE_ADDR_KEY', 'REMOTE_ADDR')
-        remote_addr = request.META.get(remote_addr_key)
+        if getattr(settings, 'RESTRICTEDSESSIONS_REMOTE_ADDR_KEY', False):
+            remote_addr_key = getattr(settings, 'RESTRICTEDSESSIONS_REMOTE_ADDR_KEY')
+            remote_addr = request.META.get(remote_addr_key)
+        else:
+            if request.META.get('HTTP_X_FORWARDED_FOR', False):
```
Comment (erikr) [inline restrictedsessions/middleware.py]: A silent usage of x-forwarded-for can be a security issue. There are situations where this may have been provided by the HTTP client, rather than set by the server. In those cases it can't be trusted.

How about simply looking at whether the value in `remote_addr` contains comma's, and if so split off the first element?

File: restrictedsessions/middleware.py [hunk 4]
```diff
@@ -80,7 +90,11 @@ def validate_ip(self, request, remote_ip):
             except AddrFormatError:
                 # session_network must be IPv4, but remote_ip is IPv6
                 return False
-        return remote_ip in session_network
+        is_ip_valid = remote_ip in session_network
```
Comment (erikr) [inline restrictedsessions/middleware.py]: I think it would be nicer to have a single place where we log about failures, so can you take this one out of this PR? If you want to improve logging, have a look at https://github.com/erikr/django-restricted-sessions/pull/10.

File: restrictedsessions/middleware.py [hunk 5]
```diff
@@ -90,4 +104,12 @@ def validate_ua(self, request):
         if SESSION_UA_KEY not in request.session:
             return True
         # Compare the new user agent value with what is known about the session
-        return request.session[SESSION_UA_KEY] == force_text(request.META['HTTP_USER_AGENT'], errors='replace')
+        ua = force_text(request.META['HTTP_USER_AGENT'], errors='replace')
```
Comment (erikr) [inline restrictedsessions/middleware.py]: Same comment as for line 93.
